[DOC] Guidelines for downgrade (#3981)
strimzi/strimzi-kafka-operator: documentation/modules/upgrading/proc-upgrade-brokers-newer-kafka.adoc

----
<1> This is changed to the new version
<2> This remains at the current version
--
+

> WARNING: You cannot downgrade Kafka once `spec.kafka.version` has changed.

Reviewer: Is this because of the `inter.broker.protocol.version`? Or why is it the case?

Reviewer 2: `inter.broker.protocol.version` indeed

Author: Changed to:

> WARNING: You cannot downgrade Kafka once `spec.kafka.version` has changed, as the `inter.broker.protocol.version` for the new Kafka version will differ.  The protocol version determines the schemas used for persistent metadata stored by the broker, including messages written to `__consumer_offsets`. The cluster will not understand the messages.

Reviewer: Are we cherry-picking this for 0.20 release branch? If not, maybe we should think about this a bit more to improve the experience of this before we do 0.21 and before we merge this.

Author: Yes. Once it's cherry-picked, we can bring the kafka downgrade sections back in and improve the chapter a whole.

----
<1> This is changed to the new version
<2> This remains at the current version
--
+

> WARNING: You cannot downgrade Kafka once `spec.kafka.version` has changed, as the `inter.broker.protocol.version` for the new Kafka version will differ. The protocol version determines the schemas used for persistent metadata stored by the broker, including messages written to `__consumer_offsets`. The cluster will not understand the messages.

Reviewer 3: ```suggestion
WARNING: You cannot downgrade Kafka after `spec.kafka.version` has changed, because the `inter.broker.protocol.version` for the new Kafka version will differ. The inter-broker protocol version determines the schemas used for persistent metadata stored by the broker, including messages written to `__consumer_offsets`. The downgraded cluster will not understand the messages.
```
